iceoryx_utils/include/iceoryx_utils/internal/units/duration.inl

```diff
@@ -21,261 +21,490 @@ namespace iox
 namespace units
 {
 template <typename T>
-constexpr Duration Duration::nanoseconds(const T ns)
+constexpr Duration Duration::nanoseconds(const T value) noexcept
 {
-    return operator"" _ns(ns);
+    static_assert(std::numeric_limits<T>::is_integer, "only integer are supported");
+
+    if (value < 0)
```

Comment:
Since this code is always the same I would create a free function to handle this error handling, otherwise we have a lot of code duplications.

What do you think of the following approach:
```cpp
template<typename T, typename OnSuccessFactory>
Duration createDurationFromValue(const T value, const OnSuccessFactory) {
    static_assert(std::numeric_limits<T>::is_integer, "only integer are supported");
    if (value < 0)
    {
        std::clog << __PRETTY_FUNCTION__ << ": Clampling negative value '" << value << "' to zero!" << std::endl;
        return Duration{0U, 0U};
    }
    return OnSuccessFactory(static_cast<unsigned long long int>(value));
}

template <typename T>
constexpr Duration Duration::nanoseconds(const T value) noexcept {
    return createDurationFromValue(value, [this](const unsigned long long int value){ return operator"" _ns(value)' });
}

```

Reply:
use a similar approach without the lambda